client/src/components/runs/logs/forms/ShareWithForm.js

```diff
@@ -169,30 +147,25 @@ export default class ShareWithForm extends React.Component {
             ...roles,
             ...(this.groupFind.value || []).map(g => g)]
         )
-      ];
+      ].sort((u1, u2) => this.sortByOverlap(u1, u2, query));
     }
-    return [...roles];
+    return [...roles].sort((u1, u2) => this.sortByOverlap(u1, u2, query));
   };
 
-  selectedUser = null;
-  selectedGroup = null;
-
   openFindUserDialog = () => {
-    this.selectedUser = null;
-    this.setState({findUserVisible: true});
+    this.setState({findUserVisible: true, userSearchString: null});
   };
 
   closeFindUserDialog = () => {
-    this.setState({findUserVisible: false});
+    this.setState({findUserVisible: false, userSearchString: null});
   };
 
   onSelectUser = async () => {
-    await this.grantPermission(this.selectedUser, true);
+    await this.grantPermission(this.state.userSearchString.trim(), true);
```

Comment:
We should check if `this.state.userSearchString` is `null`/`undefined`

Reply:
Done!